gst/nnstreamer/tensor_common_pipeline.c

```diff
@@ -445,6 +445,107 @@ gst_tensor_time_sync_buffer_from_collectpad (GstCollectPads * collect,
   return !(*is_eos);
 }
 
+/**
+ * @brief Configure gst-buffer with tensors information.
+ * NNStreamer handles single memory chunk as single tensor.
+ * If incoming buffer has invalid memories, separate it and generate new gst-buffer using tensors information.
+ * Note that this function always takes the ownership of input buffer.
+ * @param in input buffer
+ * @param config tensors config structure
+ * @return Newly allocated buffer. Null if failed. Caller should unref the buffer using gst_buffer_unref().
+ */
+GstBuffer *
+gst_tensor_buffer_from_config (GstBuffer * in, GstTensorsConfig * config)
+{
+  GstBuffer *out = NULL;
+  GstMemory *all = NULL;
+  GstMapInfo map;
+  guint i, num;
+  gsize total, offset;
+  gsize mem_size[NNS_TENSOR_SIZE_LIMIT];
+  gboolean configured = FALSE;
+
+  if (!GST_IS_BUFFER (in) || !gst_tensors_config_validate (config))
+    goto error;
```

Comment:
how about just returning NULL? `gst_buffer_unref` can be called when `in` is a non-buffer instance.

Reply:
Thanks, updated.

Reply:
We have started writing more elaborated error messages for exceptions.
Please describe the error and how the user should correct the error with this error.